### ISSUE TITLE
`vm.stopAndReturnStateDiff()` results do not include call to create2 factory

### ISSUE THREAD

**Issue by RPate97** [foundry-rs/foundry#6634]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (73fb616 2023-12-19T00:17:51.579325000Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Apple Silicon)

### Describe the bug

Deploying a contract using create2 shows up as a single create `AccountAccess` in the results returned by `vm.stopAndReturnStateDiff()` even though the call to the create2 factory itself is also a state-changing action. This makes it difficult to determine if a contract was deployed using create or create2 based on the recorded results. The call to the create2 factory should be included in the results returned by this cheat code. 

It's unclear to me if the current behavior was intended or not. If it was intended, then this issue should probably be converted into a feature request. Please let me know if you think I should do that. 

### Replication 
```
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.0;

import { Script, console } from "forge-std/Script.sol";
import { VmSafe, Vm } from "forge-std/Vm.sol";

contract Counter {
    uint256 public number;

    constructor (uint _number) {
      number = _number;
    }

    function increment() public {
      number += 1;
    }
}

contract CounterScript is Script {
    function run() public {
        vm.startStateDiffRecording();
        Counter counter = new Counter{ salt: 0 }(0);

        Vm.AccountAccess[] memory accountAccesses = vm.stopAndReturnStateDiff();
        for (uint i = 0; i < accountAccesses.length; i++) {
            console.log(uint(accountAccesses[i].kind));
            console.log('account', accountAccesses[i].account);
            console.log('accessor', accountAccesses[i].accessor);
        }
    }
}
```

`forge script ./script/Counter.s.sol --tc CounterScript`

**Comment by maurelian** [comment]
This looks to me like a legitimate issue. I notice that the trace from your PoC also excludes any reference to the factory contract and makes it appear like a direct creation. 

```
[146870] CounterScript::run()
    ├─ [0] VM::startStateDiffRecording()
    │   └─ ← ()
    ├─ [74744] → new Counter@0xa5a60F25eae05d45f90d25469954703CC5334527
    │   └─ ← 371 bytes of code
    ├─ [0] VM::stopAndReturnStateDiff()
 ```

**Comment by klkvr** [comment]
Fixed by #7653 and #7207
